sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1411,21 +1441,27 @@ static void _event_cb(netdev_t *dev, netdev_event_t event)
         gnrc_pktsnip_t *pkt = NULL;
         switch (event) {
             case NETDEV_EVENT_RX_COMPLETE:
+                _send_queued_pkt(netif);
```

Comment:
This seems problematic to me.
For one, it is completely surprising to see this here, without the context of why this PR exists in the first place.

The other problem would be that this basically requires the network device to keep the just received packet "on hold", send a queued packet, and only then get the held packet out of the device. I don't think all devices allow that.

I'm not sure, this might be solved by getting the received packet first, then initiating the send. E.g., swap 1444 and 1445. A downside would be that both packets need to be in memory at the same time, but I suspect that otherwise, the device needs to provide extra buffering...

Reply:
> The other problem would be that this basically requires the network device to keep the just received packet "on hold", send a queued packet, and only then get the held packet out of the device. I don't think all devices allow that.

Oh damn, you are right, this needs to be called after the receive.

> I'm not sure, this might be solved by getting the received packet first, then initiating the send. E.g., swap 1444 and 1445. A downside would be that both packets need to be in memory at the same time, but I suspect that otherwise, the device needs to provide extra buffering...

Yepp. Rather put it in the stack's buffer.

Reply:
> For one, it is completely surprising to see this here, without the context of why this PR exists in the first place.

How can I alleviate this surprise? Would some comment help?

Reply:
I guess so. IMO we need some documentation to point to that explains why the netif thread cannot block when it gets a send request (or why it cannot block at all actually).

For now a comment is probably sufficient. Something that refers to where the packets get queued.

Reply:
Done.